docs/excel/excel-add-ins-pivots.md

```diff
@@ -0,0 +1,216 @@
+Row and column hierarchies define how data will be grouped. For example, having a row hierarchy of **Farms** groups together all data sets from the same farm. The choice between row and column hierarchy defines the orientation of the pivot table.
+
+Data hierarchies are the values to be aggregated based on the row and column hierarchies. A pivot table with a row hierarchy of **Farms** and a data hierarchy of **Crates Sold Wholesale** shows the sum total (by default) of all the different fruits for each farm.
+
+Filter hierarchies include or exclude data from the pivot based on values within that filtered type. A filter hierarchy of **Classification** with the type **Organic** selected only shows data for organic fruit.
+
+Here is the farm data again, alongside a pivot table. The pivot table is using **Farm** and **Type** as the row hierarchies, **Crates Sold at Farm** and **Crates Sold Wholesale** as the data hierarchies (with the default aggregation function of sum), and **Classification** as a filter hierarchy (with **Organic** selected). 
+
+![A selection of fruit sales data next to a pivot table with row, data, and filter hierarchies.](../images/excel-pivot-table-and-data.png)
+
+This pivot table could be generated through the JavaScript API or through the Excel UI. Both options allow for further manipulation through add-ins.
+
+## Create a pivot table
+
+Pivot tables need a name, source, and destination. The source can be a `Range`, `string`, or `Table`. The destination can either be a `Range` or `string`. 
+The following samples show various pivot table creation techniques. All three samples feature a pivot table named **Farm Sales** created on a worksheet called **Pivot** at cell **A2**. Its data comes from the worksheet **Data** across the range **A1:E21**. 
+
+### Create a pivot table with names
+```ts
+await Excel.run(async (context) => {
+	context.workbook.worksheets.getActiveWorksheet()
+		.pivotTables.add("Farm Sales", "Data!A1:E21", "Pivot!A2");
+
+	await context.sync();
+});
+```
+
+### Create a pivot table with objects
+```ts
+await Excel.run(async (context) => {	
+	var myRange = context.workbook.worksheets.getItem("Data").getRange("A1:E21");
+	var myRange2 = context.workbook.worksheets.getItem("Pivot").getRange("A2");
+	context.workbook.worksheets.getItem("Pivot").pivotTables.add("Farm Sales", myRange, myRange2);
+	
+	await context.sync();
+});
+```
+
+### Create a pivot table from a workbook
+```ts
+await Excel.run(async (context) => {
+	context.workbook.pivotTables.add("Farm Sales", "Data!A1:E21", "Pivot!A2");
+
+	await context.sync();
+});
+```
+
+## Use an existing pivot table
+Manually created pivot tables are also accessible through the worksheet’s pivot table collection. 
+The following code gets the first pivot table in the worksheet. It then gives the table a name for easy future reference.
+
+```ts
+await Excel.run(async (context) => {
+	let pivotTableList = context.workbook.worksheets.getActiveWorksheet().pivotTables;
```

Comment:
Best practice for TypeScript is to use `const` instead of `let` for values that never get written to again after being initialized.